lib/Doctrine/Annotations/Type/Constant/ConstantStringType.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Annotations\Type\Constant;
+
+use Doctrine\Annotations\Type\ConstantScalarType;
+use Doctrine\Annotations\Type\StringType as GenericStringType;
+use function addcslashes;
+use function sprintf;
+
+/**
+ * @internal
+ */
+final class ConstantStringType extends GenericStringType implements ConstantScalarType
+{
+    /** @var string */
+    private $value;
+
+    public function __construct(string $value)
+    {
+        $this->value = $value;
+    }
+
+    public function getValue() : string
+    {
+        return $this->value;
+    }
+
+    public function describe() : string
+    {
+        return sprintf('"%s"', addcslashes($this->value, '\"'));
```

Comment:
Why not `addslashes`, but `addcslashes` instead?

Reply:
`addslashes` doesn't allow specifying characters.

Reply:
Please add comment explaining why are only these characters escaped and we purposely ignore escaping of single quote and NUL bytes.

Reply:
On second thought, it's obvious enough because it's enclosed with double quotes, so double quotes need to be escaped only